akka-projection-core/src/main/scala/akka/projection/scaladsl/ProjectionManagement.scala

```diff
@@ -51,12 +51,7 @@ import akka.util.Timeout
     })
   }
 
-  /**
-   * INTERNAL API: ProjectionBehavior registers when started.
-   */
-  @InternalApi private[projection] def register(
-      projectionId: ProjectionId,
-      projection: ActorRef[OffsetManagementCommand]): Unit = {
```

Comment:
It's `scaladsl` only. No need for the annotation/comment.

Reply:
Users can still access it if they add a class under `akka.projection` and we want to avoid it to be free to change it.

Reply:
So, what difference does it make if we keep or remove the annotation? Users writing code under the `akka.projection` package to circumvent a `private[projection]` are opening themselves to any potential API change we introduce. They were warned by the fact that the method is not public or protected. The Annotation and the comment won't change much.

My point is that `@InternalApi` is _necessary_ in scala code visible/usable from Java code, and this is not the case because it's under `scaladsl`.

Reply:
I agree that if you add a class in `akka.projection` to circumvent the `private` modifier that you are taking a risk. 

It's also quite common, but not consistent, to combine both in the Akka code base. The idea is to have an extra warning about it, I guess.

The same for classes in `internal` package. Users can use them directly, but we double down on the warning (package + annotation).